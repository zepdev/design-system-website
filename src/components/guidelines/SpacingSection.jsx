import React from "react"
import clsx from "clsx"
import { makeStyles } from "@material-ui/core/styles"
import { spacing } from "@zlab-de/zel-react/guidelines.json"

const useStyles = makeStyles(theme => ({
  root: {
    overflowX: "auto",
  },
  table: {
    width: "100%",
    marginBottom: theme.space.xl.px * 2,
    borderCollapse: "collapse",
  },
  th: {
    padding: `${theme.space.m.px}px ${theme.space.l.px}px`,
    textAlign: "left",
    display: "table-cell",
  },
  tr: {
    display: "table-row",
  },
  tbody: {
    display: "table-row-group",
  },
  header: {
    background: theme.color.global.white,
    display: "table-header-group",
  },
  lowercase: {
    textTransform: "lowercase",
  },
  capitalize: {
    textTransform: "capitalize",
  },
  borderBottom: {
    borderBottom: `1px solid ${theme.color.global.lightGray}`,
  },
  square: {
    background: theme.indigo.primary,
  },
}))

function SpacingSection() {
  const classes = useStyles()
  const header = ["name", "px", "rem", "actual size"]

  return (
    <div className={classes.root}>
      <table className={classes.table}>
        <caption className="zep-visually-hidden">Spacing Scale</caption>
        <thead className={classes.header}>
          <tr className={classes.tr}>
            {header.map(elem => (
              <th
                scope="col"
                className={clsx(classes.th, "zep-typo--normal-1", {
                  [classes.lowercase]: elem === "rem" || elem === "px",
                  [classes.capitalize]: elem !== "rem" && elem !== "px",
                })}
                key={`header_${elem}`}
              >
                {elem}
              </th>
            ))}
          </tr>
        </thead>
        <tbody className={classes.tbody}>
          {Object.keys(spacing.component).map(elem => (
            <tr key={`row_${elem}`} className={classes.tr}>
              {header.map((item, idx) => (
                <th
                  className={clsx(
                    classes.th,
                    classes.borderBottom,
                    "zep-typo--normal-body2"
                  )}
                  key={`spacing${idx}`}
                  scope={idx === 0 ? "row" : null}
                >
                  {item === "actual size" ? (
                    <div
                      style={{
                        width: spacing.component[elem].px,
                        height: spacing.component[elem].px,
                        color: "green",
                      }}
                      className={classes.square}
                    />
                  ) : item === "name" ? (
                    `zep-spacing--${elem}`
                  ) : item === "px" ? (
                    `${spacing.component[elem][item]}px`
                  ) : (
                    `${spacing.component[elem][item]}rem`
                  )}
                </th>
              ))}
            </tr>
          ))}
        </tbody>
      </table>
    </div>
  )
}

export default SpacingSection
