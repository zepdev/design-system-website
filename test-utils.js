import React from 'react'
import { render } from '@testing-library/react'
import { ThemeProvider } from '@material-ui/core/styles'
import theme from './src/theme'

const AllTheProviders = ({ children }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>
}

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }
