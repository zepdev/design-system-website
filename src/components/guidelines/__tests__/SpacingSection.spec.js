import React from "react"
import { render } from "../../../../test-utils"
import SpacingSection from "../SpacingSection"

describe("SpacingSection", () => {
  it("renders correctly", () => {
    const component = render(<SpacingSection />)
    expect(component).toMatchSnapshot()
  })
})
