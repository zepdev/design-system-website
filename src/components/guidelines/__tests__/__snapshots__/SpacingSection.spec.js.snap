// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpacingSection renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="makeStyles-root-1"
      >
        <table
          class="makeStyles-table-2"
        >
          <caption
            class="zep-visually-hidden"
          >
            Spacing Scale
          </caption>
          <thead
            class="makeStyles-header-6"
          >
            <tr
              class="makeStyles-tr-4"
            >
              <th
                class="makeStyles-th-3 zep-typo--normal-1 makeStyles-capitalize-8"
                scope="col"
              >
                name
              </th>
              <th
                class="makeStyles-th-3 zep-typo--normal-1 makeStyles-lowercase-7"
                scope="col"
              >
                px
              </th>
              <th
                class="makeStyles-th-3 zep-typo--normal-1 makeStyles-lowercase-7"
                scope="col"
              >
                rem
              </th>
              <th
                class="makeStyles-th-3 zep-typo--normal-1 makeStyles-capitalize-8"
                scope="col"
              >
                actual size
              </th>
            </tr>
          </thead>
          <tbody
            class="makeStyles-tbody-5"
          >
            <tr
              class="makeStyles-tr-4"
            >
              <th
                class="makeStyles-th-3 makeStyles-borderBottom-9 zep-typo--normal-body2"
                scope="row"
              >
                zep-spacing--xxs
              </th>
              <th
                class="makeStyles-th-3 makeStyles-borderBottom-9 zep-typo--normal-body2"
              >
                4px
              </th>
              <th
                class="makeStyles-th-3 makeStyles-borderBottom-9 zep-typo--normal-body2"
              >
                0.25rem
              </th>
              <th
                class="makeStyles-th-3 makeStyles-borderBottom-9 zep-typo--normal-body2"
              >
                <div
                  class="makeStyles-square-10"
                  style="width: 4px; height: 4px; color: green;"
                />
              </th>
            </tr>
            <tr
              class="makeStyles-tr-4"
            >
              <th
                class="makeStyles-th-3 makeStyles-borderBottom-9 zep-typo--normal-body2"
                scope="row"
              >
                zep-spacing--xs
              </th>
              <th
                class="makeStyles-th-3 makeStyles-borderBottom-9 zep-typo--normal-body2"
              >
                8px
              </th>
              <th
                class="makeStyles-th-3 makeStyles-borderBottom-9 zep-typo--normal-body2"
              >
                0.5rem
              </th>
              <th
                class="makeStyles-th-3 makeStyles-borderBottom-9 zep-typo--normal-body2"
              >
                <div
                  class="makeStyles-square-10"
                  style="width: 8px; height: 8px; color: green;"
                />
              </th>
            </tr>
            <tr
              class="makeStyles-tr-4"
            >
              <th
                class="makeStyles-th-3 makeStyles-borderBottom-9 zep-typo--normal-body2"
                scope="row"
              >
                zep-spacing--s
              </th>
              <th
                class="makeStyles-th-3 makeStyles-borderBottom-9 zep-typo--normal-body2"
              >
                12px
              </th>
              <th
                class="makeStyles-th-3 makeStyles-borderBottom-9 zep-typo--normal-body2"
              >
                0.75rem
              </th>
              <th
                class="makeStyles-th-3 makeStyles-borderBottom-9 zep-typo--normal-body2"
              >
                <div
                  class="makeStyles-square-10"
                  style="width: 12px; height: 12px; color: green;"
                />
              </th>
            </tr>
            <tr
              class="makeStyles-tr-4"
            >
              <th
                class="makeStyles-th-3 makeStyles-borderBottom-9 zep-typo--normal-body2"
                scope="row"
              >
                zep-spacing--m
              </th>
              <th
                class="makeStyles-th-3 makeStyles-borderBottom-9 zep-typo--normal-body2"
              >
                16px
              </th>
              <th
                class="makeStyles-th-3 makeStyles-borderBottom-9 zep-typo--normal-body2"
              >
                1rem
              </th>
              <th
                class="makeStyles-th-3 makeStyles-borderBottom-9 zep-typo--normal-body2"
              >
                <div
                  class="makeStyles-square-10"
                  style="width: 16px; height: 16px; color: green;"
                />
              </th>
            </tr>
            <tr
              class="makeStyles-tr-4"
            >
              <th
                class="makeStyles-th-3 makeStyles-borderBottom-9 zep-typo--normal-body2"
                scope="row"
              >
                zep-spacing--l
              </th>
              <th
                class="makeStyles-th-3 makeStyles-borderBottom-9 zep-typo--normal-body2"
              >
                24px
              </th>
              <th
                class="makeStyles-th-3 makeStyles-borderBottom-9 zep-typo--normal-body2"
              >
                1.5rem
              </th>
              <th
                class="makeStyles-th-3 makeStyles-borderBottom-9 zep-typo--normal-body2"
              >
                <div
                  class="makeStyles-square-10"
                  style="width: 24px; height: 24px; color: green;"
                />
              </th>
            </tr>
            <tr
              class="makeStyles-tr-4"
            >
              <th
                class="makeStyles-th-3 makeStyles-borderBottom-9 zep-typo--normal-body2"
                scope="row"
              >
                zep-spacing--xl
              </th>
              <th
                class="makeStyles-th-3 makeStyles-borderBottom-9 zep-typo--normal-body2"
              >
                32px
              </th>
              <th
                class="makeStyles-th-3 makeStyles-borderBottom-9 zep-typo--normal-body2"
              >
                2rem
              </th>
              <th
                class="makeStyles-th-3 makeStyles-borderBottom-9 zep-typo--normal-body2"
              >
                <div
                  class="makeStyles-square-10"
                  style="width: 32px; height: 32px; color: green;"
                />
              </th>
            </tr>
            <tr
              class="makeStyles-tr-4"
            >
              <th
                class="makeStyles-th-3 makeStyles-borderBottom-9 zep-typo--normal-body2"
                scope="row"
              >
                zep-spacing--xxl
              </th>
              <th
                class="makeStyles-th-3 makeStyles-borderBottom-9 zep-typo--normal-body2"
              >
                48px
              </th>
              <th
                class="makeStyles-th-3 makeStyles-borderBottom-9 zep-typo--normal-body2"
              >
                3rem
              </th>
              <th
                class="makeStyles-th-3 makeStyles-borderBottom-9 zep-typo--normal-body2"
              >
                <div
                  class="makeStyles-square-10"
                  style="width: 48px; height: 48px; color: green;"
                />
              </th>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="makeStyles-root-1"
    >
      <table
        class="makeStyles-table-2"
      >
        <caption
          class="zep-visually-hidden"
        >
          Spacing Scale
        </caption>
        <thead
          class="makeStyles-header-6"
        >
          <tr
            class="makeStyles-tr-4"
          >
            <th
              class="makeStyles-th-3 zep-typo--normal-1 makeStyles-capitalize-8"
              scope="col"
            >
              name
            </th>
            <th
              class="makeStyles-th-3 zep-typo--normal-1 makeStyles-lowercase-7"
              scope="col"
            >
              px
            </th>
            <th
              class="makeStyles-th-3 zep-typo--normal-1 makeStyles-lowercase-7"
              scope="col"
            >
              rem
            </th>
            <th
              class="makeStyles-th-3 zep-typo--normal-1 makeStyles-capitalize-8"
              scope="col"
            >
              actual size
            </th>
          </tr>
        </thead>
        <tbody
          class="makeStyles-tbody-5"
        >
          <tr
            class="makeStyles-tr-4"
          >
            <th
              class="makeStyles-th-3 makeStyles-borderBottom-9 zep-typo--normal-body2"
              scope="row"
            >
              zep-spacing--xxs
            </th>
            <th
              class="makeStyles-th-3 makeStyles-borderBottom-9 zep-typo--normal-body2"
            >
              4px
            </th>
            <th
              class="makeStyles-th-3 makeStyles-borderBottom-9 zep-typo--normal-body2"
            >
              0.25rem
            </th>
            <th
              class="makeStyles-th-3 makeStyles-borderBottom-9 zep-typo--normal-body2"
            >
              <div
                class="makeStyles-square-10"
                style="width: 4px; height: 4px; color: green;"
              />
            </th>
          </tr>
          <tr
            class="makeStyles-tr-4"
          >
            <th
              class="makeStyles-th-3 makeStyles-borderBottom-9 zep-typo--normal-body2"
              scope="row"
            >
              zep-spacing--xs
            </th>
            <th
              class="makeStyles-th-3 makeStyles-borderBottom-9 zep-typo--normal-body2"
            >
              8px
            </th>
            <th
              class="makeStyles-th-3 makeStyles-borderBottom-9 zep-typo--normal-body2"
            >
              0.5rem
            </th>
            <th
              class="makeStyles-th-3 makeStyles-borderBottom-9 zep-typo--normal-body2"
            >
              <div
                class="makeStyles-square-10"
                style="width: 8px; height: 8px; color: green;"
              />
            </th>
          </tr>
          <tr
            class="makeStyles-tr-4"
          >
            <th
              class="makeStyles-th-3 makeStyles-borderBottom-9 zep-typo--normal-body2"
              scope="row"
            >
              zep-spacing--s
            </th>
            <th
              class="makeStyles-th-3 makeStyles-borderBottom-9 zep-typo--normal-body2"
            >
              12px
            </th>
            <th
              class="makeStyles-th-3 makeStyles-borderBottom-9 zep-typo--normal-body2"
            >
              0.75rem
            </th>
            <th
              class="makeStyles-th-3 makeStyles-borderBottom-9 zep-typo--normal-body2"
            >
              <div
                class="makeStyles-square-10"
                style="width: 12px; height: 12px; color: green;"
              />
            </th>
          </tr>
          <tr
            class="makeStyles-tr-4"
          >
            <th
              class="makeStyles-th-3 makeStyles-borderBottom-9 zep-typo--normal-body2"
              scope="row"
            >
              zep-spacing--m
            </th>
            <th
              class="makeStyles-th-3 makeStyles-borderBottom-9 zep-typo--normal-body2"
            >
              16px
            </th>
            <th
              class="makeStyles-th-3 makeStyles-borderBottom-9 zep-typo--normal-body2"
            >
              1rem
            </th>
            <th
              class="makeStyles-th-3 makeStyles-borderBottom-9 zep-typo--normal-body2"
            >
              <div
                class="makeStyles-square-10"
                style="width: 16px; height: 16px; color: green;"
              />
            </th>
          </tr>
          <tr
            class="makeStyles-tr-4"
          >
            <th
              class="makeStyles-th-3 makeStyles-borderBottom-9 zep-typo--normal-body2"
              scope="row"
            >
              zep-spacing--l
            </th>
            <th
              class="makeStyles-th-3 makeStyles-borderBottom-9 zep-typo--normal-body2"
            >
              24px
            </th>
            <th
              class="makeStyles-th-3 makeStyles-borderBottom-9 zep-typo--normal-body2"
            >
              1.5rem
            </th>
            <th
              class="makeStyles-th-3 makeStyles-borderBottom-9 zep-typo--normal-body2"
            >
              <div
                class="makeStyles-square-10"
                style="width: 24px; height: 24px; color: green;"
              />
            </th>
          </tr>
          <tr
            class="makeStyles-tr-4"
          >
            <th
              class="makeStyles-th-3 makeStyles-borderBottom-9 zep-typo--normal-body2"
              scope="row"
            >
              zep-spacing--xl
            </th>
            <th
              class="makeStyles-th-3 makeStyles-borderBottom-9 zep-typo--normal-body2"
            >
              32px
            </th>
            <th
              class="makeStyles-th-3 makeStyles-borderBottom-9 zep-typo--normal-body2"
            >
              2rem
            </th>
            <th
              class="makeStyles-th-3 makeStyles-borderBottom-9 zep-typo--normal-body2"
            >
              <div
                class="makeStyles-square-10"
                style="width: 32px; height: 32px; color: green;"
              />
            </th>
          </tr>
          <tr
            class="makeStyles-tr-4"
          >
            <th
              class="makeStyles-th-3 makeStyles-borderBottom-9 zep-typo--normal-body2"
              scope="row"
            >
              zep-spacing--xxl
            </th>
            <th
              class="makeStyles-th-3 makeStyles-borderBottom-9 zep-typo--normal-body2"
            >
              48px
            </th>
            <th
              class="makeStyles-th-3 makeStyles-borderBottom-9 zep-typo--normal-body2"
            >
              3rem
            </th>
            <th
              class="makeStyles-th-3 makeStyles-borderBottom-9 zep-typo--normal-body2"
            >
              <div
                class="makeStyles-square-10"
                style="width: 48px; height: 48px; color: green;"
              />
            </th>
          </tr>
        </tbody>
      </table>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
